ui/src/components/PathList/PathListItem.tsx

```diff
@@ -0,0 +1,129 @@
+import { Accordion, AccordionSummary, AccordionDetails, Grid } from "@mui/material";
+import { useContext, useMemo, useEffect } from "react";
+import { CollapsedContext } from "../../CollapsedContext";
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import './PathListItem.sass';
+
+export interface PathListItemProps {
+    change: any
+    showChangeType?: string
+}
+
+export const PathListItem: React.FC<PathListItemProps> = ({ change, showChangeType = "" }) => {
+    const changeVal = change.Value
+    const { accordions, setAccordions } = useContext(CollapsedContext);
+    const filteredChanges = useMemo(() => {
+        const changes = showChangeType ? changeVal?.Paths.filter((path: any) => path.Changelog.type === showChangeType) : changeVal?.Paths
+        setAccordions((prev) => {
+            let newArr = [...prev, changes.map((path: any) => { return { isCollpased: true, id: JSON.stringify(path) } })].flat()
+            return newArr
+        })
+        return changes
+    }, [changeVal?.Paths, setAccordions, showChangeType])
+
+    const onClick = (id: string) => {
+        setAccordions((prev) => {
+            let newArr = [...prev]
+            let accordion = newArr.find(x => x.id === id)
+            if (accordion)
+                accordion.isCollpased = !accordion?.isCollpased
+            return newArr
+        })
+    }
+
+    useEffect(() => {
+        setAccordions(prev => [...prev, { isCollpased: true, id: JSON.stringify(change) }])
+    }, [change, setAccordions])
+
+    const isExpand = (path: any) => {
+        const acc = accordions?.find(x => x.id === JSON.stringify(path))
+        if (acc)
+            return !acc.isCollpased
+        return false
+    }
+
+    const getToTypeColor = (type: string) => {
+        switch (type) {
+            case "create":
+                return "create"
+            case "update":
+                return "create"
+            case "delete":
+                return "delete"
+            default:
+                return "info"
+        }
+    }
+
+    const getFromTypeColor = (type: string) => {
+        switch (type) {
+            case "create":
+                return "create"
+            case "update":
+                return "delete"
+            case "delete":
+                return "delete"
+            default:
+                return "info"
+        }
+    }
+
+    return (
+        <Accordion expanded={!accordions.find(x => x.id === JSON.stringify(change))?.isCollpased}>
+            <AccordionSummary
+                expandIcon={<ExpandMoreIcon />}
+                aria-controls="panel2a-content" onClick={() => onClick(JSON.stringify(change))}>
+                <div className='accordionTitle'>
+                    <div className='path'>
+                        <span className='pathPrefix'>{change.Value.Key}</span>&nbsp;
```

Comment:
can we replace &nbsp; with css?

Reply:
what do you mean?
if I understand correcly, since it is part of the structure and not design, its should be in the HTML IMO

Reply:
I mean to the nbsp char